packages/sms-language-server/src/SmsLanguageServer.ts

```diff
@@ -53,9 +68,94 @@ export class SmsLanguageServer extends AbstractLanguageServer<SmsParser> {
     });
   }
 
-  handleCompletion(
-    params: lsp.TextDocumentPositionParams
-  ): lsp.CompletionItem[] {
+  getIndentFoldingRange(lines, lineIdx) {
+    const start = lineIdx;
+    const startingLine = lines[lineIdx];
+    const startIndentLevel =
+      startingLine.length - startingLine.trimLeft().length;
+
+    for (let i = lineIdx + 1; i < lines.length; i++) {
+      const lowerCaseLine = lines[i].toLowerCase();
+      const trimmedLine = lowerCaseLine.trimLeft();
+      const indentLevel = lowerCaseLine.length - trimmedLine.length;
+
+      if (indentLevel <= startIndentLevel) {
+        return {
+          startLine: start,
+          endLine: i - 1,
+          kind: 'indent',
+        };
+      }
+    }
+    return {
+      startLine: start,
+      endLine: lines.length - 1,
+      kind: 'indent',
+    };
+  }
+
+  getPrefixFoldingRange(lines, lineIdx) {
+    const startLine = lineIdx;
+
+    for (let i = lineIdx + 1; i < lines.length; i++) {
+      const lowerCaseLine = lines[i].toLowerCase().trimLeft();
+      if (
+        !lowerCaseLine.startsWith('prefix') &&
+        !lowerCaseLine.startsWith('@prefix')
```

Comment:
SMS2 supports 'prefix' but I don't think it supports '@prefix' (since it's based on SPARQL rather than Turtle). Our SmsParser at least does not allow '@prefix'.